### ISSUE TITLE
Trigger event-handler on node itself upon join or leave

### ISSUE THREAD

**Issue by ssenaria** [hashicorp/serf#358]
I have an event handler that runs when a node either joins or leaves the cluster. What I didn't realize is that it also runs on the other nodes. I understand when a node leaves, the node itself wouldn't be able to run the event handler and other nodes in the cluster will run. However, the script i have for member-join, I would prefer that it runs on the node that's joining and nowhere else.

**Comment by slackpad** [comment]
Hi @ssenaria can you make use of https://www.serfdom.io/docs/agent/event-handlers.html#SERF_SELF_NAME to do some filtering to achieve this (ie. all nodes run the event but only the newly-added node makes it past the filter and runs the rest of the script)?

**Comment by ssenaria** [comment]
Maybe I'm using it wrong but here's what I'm seeing. BTW, I do have a work around for this, it's kinda of funky but it works. If there's a more efficient way, I would like to try that. ANyways, using `SERF_SELF_NAME` yields the following.

I have two nodes. I failed one and minutes later bring it back into the cluster. I saved STDIN into a file and I see the following. I'm not sure how I deduce from that and create a filter that would allow me to know which node was recently added.  What I ended up doing was grep for the number of lines and if it's more than 1, it would be the host I want.

On failed node :

    ocdbeta.mydomain	10.199.1.50 cluster_group=odd,datacenter=phl,environment_id=sf_phl,host_type=web
    myweb001.mydomain	10.199.1.99 host_type=batch,cluster_group=odd,datacenter=phl,environment_id=sf_phl
    SERF_SELF_NAME = ocdbeta.mydomain

On surviving node:

    ocdbeta.mydomain	10.199.1.50 cluster_group=odd,datacenter=phl,environment_id=sf_phl,host_type=web
    SERF_SELF_NAME = myweb001.mydomain
